Notify user of dependent `search_engine` parameter

harmstyler opened the pull request:
The `search_engine` doesn't exist by default and is required for the new search engine to function.

harmstyler committed:
Notify user of dependent `search_engine` parameter

The `search_engine` doesn't exist by default and is required for the new search engine to function.

emodric commented:
Hi!

Are you sure this is needed?

`search_engine` parameter should exist as it is specified in eZ Platform `parameters.yml.dist`

https://github.com/ezsystems/ezplatform/blob/master/ezpublish/config/parameters.yml.dist#L23

harmstyler commented:
I am running eZ Publish 5.4, not eZ Platform, and it appears that my install does not have that parameter by default. It is also not mentioned at all in the [Solr Search Engine Bundle doc](https://doc.ez.no/display/EZP/Solr+Search+Engine+Bundle) as far as I can see. If it is going to be a default in eZ Platform it is probably not required in the install docs but I feel it should be somewhere (maybe a note if not running eZ Platform). It wasn't obvious to me what the parameter should be set to until I browsed the class to find its usage.

emodric commented:
Okay, that makes sense :)

Can you modify your text to say something in line of: Check if you have a `search_engine` parameter in your `parameters.yml` file, and if you don't set it to whatever search engine you want to use... (It doesn't have to be `solr`, it can be `legacy` too, so I would avoid mentioning the specific engine)

harmstyler committed:
Update to verbiage of the `search_engine` param

The `search_engine` parameter is set by default in eZ Platform, so some installs might already have the param set.

harmstyler commented:
I updated the verbiage to make sure the user both knows the parameter is required and they might not have it set yet.

emodric commented:
Thanks!
